docs/source/guide/distributed-partition.rst

```diff
@@ -0,0 +1,307 @@
+
+.. code-block:: none
+
+    0 1 0 0
+    0 1 0 1
+    0 1 0 2
+    1 0 1 0
+    1 0 1 1
+    1 0 1 2
+    1 0 1 3
+
+Similarly, each row in ``xxx_edges.txt`` stores the information of an edge with the following format:
+
+.. code-block:: none
+
+    <src_id> <dst_id> <type_edge_id> <edge_type>
+
+All fields are separated by whitespace:
+
+* ``<src_id>`` is the *homogeneous* ID of the source node.
+* ``<dst_id>`` is the *homogeneous* ID of the destination node.
+* ``<type_edge_id>`` is the edge ID for the edge type.
+* ``<edge_type>`` is the edge type.
+
+``xxx_stats.txt`` stores some basic statistics of the graph. It has only one line with three fields
+separated by whitespace:
+
+.. code-block:: none
+
+    <num_nodes> <num_edges> <num_node_weights>
+
+* ``num_nodes`` stores the total number of nodes regardless of node types.
+* ``num_edges`` stores the total number of edges regardless of edge types.
+* ``num_node_weights`` stores the number of node weights in the node file.
+
+Run ParMETIS and output format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ParMETIS contains a command called ``pm_dglpart``, which loads the graph stored
+in the three files from the machine where ``pm_dglpart`` is invoked, distributes
+data to all machines in the cluster and invokes ParMETIS to partition the
+graph. When it completes, it generates three files for each partition:
+``p<part_id>-xxx_nodes.txt``, ``p<part_id>-xxx_edges.txt``,
+``p<part_id>-xxx_stats.txt``.
+
+.. note::
+
+    ParMETIS reassigns IDs to nodes during the partitioning. After ID reassignment,
+    the nodes in a partition are assigned with contiguous IDs; furthermore, the nodes of
+    the same type are assigned with contiguous IDs.
+
+``p<part_id>-xxx_nodes.txt`` stores the node data of the partition. Each row represents
+a node with the following fields:
+
+.. code-block:: none
+
+    <node_id> <node_type> <weight1> ... <orig_type_node_id>
+
+* ``<node_id>`` is the *homogeneous* node IDs after ID reassignment.
+* ``<node_type>`` is the node type.
+* ``<weight1>`` is the node weight used by ParMETIS.
+* ``<orig_type_node_id>`` is the original node ID for a specific node type in the input heterogeneous graph.
+* ``<attributes>`` are optional fields that contain any node attributes in the input node file.
+
+``p<part_id>-xxx_edges.txt`` stores the edge data of the partition. Each row represents
+an edge with the following fields:
+
+.. code-block:: none
+
+    <src_id> <dst_id> <orig_src_id> <orig_dst_id> <orig_type_edge_id> <edge_type>
+
+* ``<src_id>`` is the *homogeneous* ID of the source node after ID reassignment.
+* ``<dst_id>`` is the *homogeneous* ID of the destination node after ID reassignment.
+* ``<orig_src_id>`` is the *homogeneous* ID of the source node in the input graph.
+* ``<orig_dst_id>`` is the *homogeneous* ID of the destination node in the input graph.
+* ``<orig_type_edge_id>`` is the edge ID for the specific edge type in the input graph.
```

Comment:
Is the edge ID here the one of the specific edge type in the input heterogeneous graph?

Reply:
I.E. per-type edge ID.

Reply:
yes. I changed it to type_wise_edge_id